grakn-graql/src/main/java/ai/grakn/graql/internal/query/analytics/AbstractClusterQuery.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016-2018 Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package ai.grakn.graql.internal.query.analytics;
+
+import ai.grakn.graql.ComputeQuery;
+
+import javax.annotation.CheckReturnValue;
+
+abstract class AbstractClusterQuery<T, V extends ComputeQuery<T>>
+        extends AbstractComputeQuery<T, V> {
+    /**
+     * The centrality measures supported.
+     */
+    enum ClusterMeasure {
```

Comment:
Maybe we should use `String`s, unless we have another idea about how to use this